flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -339,15 +348,29 @@ public Builder setSnapshotProperty(String property, String value) {
      */
     @Deprecated
     public DataStreamSink<RowData> build() {
-      return chainIcebergOperators();
+      return append();
     }
 
     /**
      * Append the iceberg sink operators to write records to iceberg table.
      *
      * @return {@link DataStreamSink} for sink.
      */
-    public DataStreamSink<Void> append() {
+    @SuppressWarnings("rawtypes")
+    public DataStreamSink append() {
+      Preconditions.checkNotNull(tableLoader, "Table loader shouldn't be null");
+
+      if (table == null) {
+        tableLoader.open();
+        try (TableLoader loader = tableLoader) {
+          this.table = loader.loadTable();
+        } catch (IOException e) {
+          throw new UncheckedIOException("Failed to load iceberg table from table loader: " + tableLoader, e);
+        }
+      }
+
+      props = buildProperties(table);
```

Comment:
It seems to be unnecessary to define this `buildProperties` method, how about just moving its body into this `append` method ?

Reply:
ok, let me edit it.